src/BizHawk.Emulation.Cores/Waterbox/ElfLoader.cs

```diff
@@ -0,0 +1,356 @@
+			if (_elf.Sections.Any(s => s.Name.StartsWith(".rel")))
+				throw new InvalidOperationException("Elf has relocations!");
+
+			_allSymbols = ((ISymbolTable)_elf.GetSection(".symtab"))
+				.Entries
+				.Cast<SymbolEntry<ulong>>()
+				.ToList();
+
+			_sectionsByName = _elf.Sections
+				.ToDictionary(s => s.Name);
+			
+			_sectionsByName.TryGetValue(".wbxsyscall", out _imports);
+			_sectionsByName.TryGetValue(".sealed", out _sealed);
+			_sectionsByName.TryGetValue(".invis", out _invisible);
+
+			_savedSections = _elf.Sections
+				.Where(s => (s.Flags & SectionFlags.Allocatable) != 0 && (s.Flags & SectionFlags.Writable) != 0)
+				.Where(s => !IsSpecialReadonlySection(s) && s != _invisible)
+				.OrderBy(s => s.LoadAddress)
+				.ToList();
+
+			_visibleSymbols = _allSymbols
+				.Where(s => s.Binding == SymbolBinding.Global && s.Visibility == SymbolVisibility.Default)
+				.ToDictionary(s => s.Name);
+
+			_importSymbols = _allSymbols
+				// TODO: No matter what attributes I provide, I seem to end up with Local and/or Hidden symbols in
+				// .wbxsyscall a lot of the time on heavily optimized release builds.
+				// Fortunately, there's nothing else in .wbxsyscall so we can just not filter at all.
+				.Where(s => s.PointedSection == _imports)
+				.ToList();
+
+
+			Memory = MemoryBlock.CallPlatformCtor(start, size);
+			Memory.Activate();
+			Memory.Protect(Memory.Start, Memory.Size, MemoryBlockBase.Protection.RW);
+
+			foreach (var seg in loadsegs)
+			{
+				var data = seg.GetFileContents();
+				Marshal.Copy(data, 0, Z.US(seg.Address), Math.Min((int)seg.Size, (int)seg.FileSize));
+			}
+
+			PrintSections();
+			PrintGdbData();
+			PrintTopSavableSymbols();
+			Protect();
+		}
+
+		private void PrintGdbData()
+		{
+			Console.WriteLine("GDB Symbol Load:");
+			Console.WriteLine($"  add-sym {ModuleName}");
+		}
+
+		private void PrintSections()
+		{
+			Console.WriteLine($"Mounted `{ModuleName}` @{Memory.Start:x16}");
+			foreach (var s in _elf.Sections.OrderBy(s => s.LoadAddress))
+			{
+				Console.WriteLine("  @{0:x16} {1}{2}{3}{4} `{5}` {6} bytes",
+					s.LoadAddress,
+					(s.Flags & SectionFlags.Allocatable) != 0 ? "R" : " ",
+					(s.Flags & SectionFlags.Writable) != 0 ? "W" : " ",
+					(s.Flags & SectionFlags.Executable) != 0 ? "X" : " ",
+					_savedSections.Contains(s) ? "V" : " ",
+					s.Name,
+					s.Size);
+			}
+		}
+
+		private void PrintTopSavableSymbols()
+		{
+			Console.WriteLine("Top savestate symbols:");
+			var tops = _allSymbols
+				.Where(s => _savedSections.Contains(s.PointedSection))
+				.OrderByDescending(s => s.Size)
+				.Where(s => s.Size >= 20 * 1024)
+				.Take(30)
+				.Select(s => $"  {s.Name} {s.Size / 1024}kiB");
+
+			foreach (var text in tops)
+			{
+				Console.WriteLine(text);
+			}
+		}
+
+		/// <summary>
+		/// Returns true if section is readonly after init
+		/// </summary>
+		private bool IsSpecialReadonlySection(Section<ulong> sec)
+		{
+			return sec.Name.Contains(".rel.ro")

```

Comment:
Revisit why this section exists.  Is there any reason to compile libc with fPIC?